### ISSUE TITLE
1.0.3 new account & anonymous can't see any chat channels and new account can't login

### ISSUE THREAD

**Issue by online0227** [RocketChat/Rocket.Chat#14497]
### Description:
New accounts and anonymous are not being able to see any chat channels. (for anonymous, it was fixed in 1.0.2 but in 1.0.3 still that bug exists. Plus now newly created user cannot see any channels after logging in as well after verifying email. Also it does not allow to login with new account). 

### Steps to reproduce:

1.  installed via snap from previous version and it got automatically updated to 1.0.3
2.  just clicked chat anonymous, but all channels are invisible.
3.  or sign up with new account and try to login, but all channels are invisible.

### Expected behavior:
anonymouse login then set username, chat.

### Actual behavior:
all channel unvisible once after cliking talk anonymous

### Server Setup Information:

- Version of Rocket.Chat Server: 1.0.3
- Operating System:  ubuntu 18.04 google cloud engine
- Deployment Method: 
- Number of Running Instances: 
- DB Replicaset Oplog: 
- NodeJS Version: 
- MongoDB Version: 

### Relevant logs:

snap is automatically updated with 1.0.3 but I found the problems so I just leave this message.

**Comment by online0227** [comment]
I just found that new account also can't see any chat channels after logging in just like anonymous.

**Comment by Cymity** [comment]
I can confirm I have the same problem with a manual installation of 1.0.3

### The issue
Anonymous users and people trying to register to the instance are unable to see any channels and have visually broken UI.

### What have I done so far?
The rocketchat logs weren't spitting out any explicit errors and I had the 

I dropped the rocketchat database in MongoDB, along with completely uninstalling and removing every file relating to MongoDB. I also deleted everything related to Node and removed the rocketchat directories.

Initially I thought it was a problem with MongoDB 4.0 so I used 3.6 instead after doing the steps above but the same error still occured.

### What I've discovered
Technically (whether anonymous or registered through email) the account was being created. You are able to proceed to the "My Profile" page and browse around there but that is about it.

The username "@" was not being created. After manually entering a username in the field and saving changes, the user is then able to browse around all the channels freely.

I also discovered that this issue only happens when you set "Allow Anonymous Read/Write" to True.
If disabled, the registering user is able to browse the channels instantly and the username @ is created, but of course with the loss of the anonymous functionality.

### Logs
Registering an account with the anonymous feature on
```
I20190517-20:40:34.073(2) Meteor ➔ method registerUser -> userId: null , arguments:  [ { name: 'testy',     email: 'testy@testssyty.com',     pass: 'testy',     'confirm-pass': 'testy' } ] 
I20190517-20:40:34.398(2) ====== BEGIN MAIL #7 ====== 
Mail Jargon Deleted 
I20190517-20:40:34.428(2) ====== END MAIL #7 ====== 
I20190517-20:40:34.633(2) Meteor ➔ publish null -> userId: 8MnTq6JtuauRQ86iC , arguments:  [] 
I20190517-20:40:34.723(2) Meteor ➔ publish stream-importers -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 'progress', { useCollection: false, args: [] } ] 
I20190517-20:40:34.740(2) Meteor ➔ method autoTranslate.getSupportedLanguages -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 'en' ] 
I20190517-20:40:34.747(2) API ➔ debug GET: /api/v1/commands.list 
I20190517-20:40:34.755(2) Meteor ➔ method listCustomSounds -> userId: 8MnTq6JtuauRQ86iC , arguments:  [] 
I20190517-20:40:34.779(2) API ➔ debug Success { statusCode: 200,   body:     { commands:        [ [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object] ],      offset: 0,      count: 23,      total: 23,      success: true } } 
I20190517-20:40:34.787(2) Meteor ➔ method listEmojiCustom -> userId: 8MnTq6JtuauRQ86iC , arguments:  [] 
I20190517-20:40:34.796(2) Meteor ➔ method apps/is-enabled -> userId: 8MnTq6JtuauRQ86iC , arguments:  [] 
I20190517-20:40:34.801(2) Meteor ➔ publish webdavAccounts -> userId: 8MnTq6JtuauRQ86iC , arguments:  [] 
I20190517-20:40:34.809(2) Meteor ➔ method getUserRoles -> userId: 8MnTq6JtuauRQ86iC , arguments:  [] 
I20190517-20:40:34.827(2) Meteor ➔ publish userData -> userId: 8MnTq6JtuauRQ86iC , arguments:  [] 
I20190517-20:40:34.836(2) Meteor ➔ publish activeUsers -> userId: 8MnTq6JtuauRQ86iC , arguments:  [] 
I20190517-20:40:34.847(2) Meteor ➔ method rooms/get -> userId: 8MnTq6JtuauRQ86iC , arguments:  [] 
I20190517-20:40:34.857(2) Meteor ➔ method subscriptions/get -> userId: 8MnTq6JtuauRQ86iC , arguments:  [] 
I20190517-20:40:34.885(2) API ➔ debug GET: /api/apps/languages 
I20190517-20:40:34.887(2) API ➔ debug Success { statusCode: 200, body: { apps: [], success: true } } 
I20190517-20:40:34.891(2) Meteor ➔ publish stream-apps -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 'app/added', { useCollection: false, args: [] } ] 
I20190517-20:40:34.895(2) Meteor ➔ publish stream-apps -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 'app/removed', { useCollection: false, args: [] } ] 
I20190517-20:40:34.899(2) Meteor ➔ publish stream-apps -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 'app/updated', { useCollection: false, args: [] } ] 
I20190517-20:40:34.907(2) Meteor ➔ publish stream-apps -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 'app/statusUpdate', { useCollection: false, args: [] } ] 
I20190517-20:40:34.911(2) Meteor ➔ publish stream-apps -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 'app/settingUpdated', { useCollection: false, args: [] } ] 
I20190517-20:40:34.914(2) Meteor ➔ publish stream-apps -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 'command/added', { useCollection: false, args: [] } ] 
I20190517-20:40:34.917(2) Meteor ➔ publish stream-apps -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 'command/disabled', { useCollection: false, args: [] } ] 
I20190517-20:40:34.925(2) Meteor ➔ publish stream-apps -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 'command/updated', { useCollection: false, args: [] } ] 
I20190517-20:40:34.928(2) Meteor ➔ publish stream-apps -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 'command/removed', { useCollection: false, args: [] } ] 
I20190517-20:40:34.931(2) Meteor ➔ method userSetUtcOffset -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ 1 ] 
I20190517-20:40:36.663(2) Meteor ➔ method UserPresence:online -> userId: 8MnTq6JtuauRQ86iC , arguments:  [ null ]
```

Registering an anonymous account with the anonymous feature enabled
```
I20190517-20:42:11.409(2) Meteor ➔ method registerUser -> userId: null , arguments:  [ {} ] 
I20190517-20:42:11.518(2) Meteor ➔ publish null -> userId: guHukrC4WR8wSTqPL , arguments:  [] 
I20190517-20:42:11.644(2) API ➔ debug GET: /api/v1/commands.list 
I20190517-20:42:11.661(2) API ➔ debug Success { statusCode: 200,   body:     { commands:        [ [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object] ],      offset: 0,      count: 23,      total: 23,      success: true } } 
I20190517-20:42:11.665(2) Meteor ➔ publish stream-importers -> userId: guHukrC4WR8wSTqPL , arguments:  [ 'progress', { useCollection: false, args: [] } ] 
I20190517-20:42:11.672(2) Meteor ➔ method autoTranslate.getSupportedLanguages -> userId: guHukrC4WR8wSTqPL , arguments:  [ 'en' ] 
I20190517-20:42:11.677(2) Meteor ➔ method listCustomSounds -> userId: guHukrC4WR8wSTqPL , arguments:  [] 
I20190517-20:42:11.685(2) Meteor ➔ method listEmojiCustom -> userId: guHukrC4WR8wSTqPL , arguments:  [] 
I20190517-20:42:11.689(2) Meteor ➔ method apps/is-enabled -> userId: guHukrC4WR8wSTqPL , arguments:  [] 
I20190517-20:42:11.693(2) Meteor ➔ publish webdavAccounts -> userId: guHukrC4WR8wSTqPL , arguments:  [] 
I20190517-20:42:11.704(2) Meteor ➔ method getUserRoles -> userId: guHukrC4WR8wSTqPL , arguments:  [] 
I20190517-20:42:11.718(2) Meteor ➔ publish userData -> userId: guHukrC4WR8wSTqPL , arguments:  [] 
I20190517-20:42:11.729(2) Meteor ➔ publish activeUsers -> userId: guHukrC4WR8wSTqPL , arguments:  [] 
I20190517-20:42:11.734(2) Meteor ➔ method rooms/get -> userId: guHukrC4WR8wSTqPL , arguments:  [] 
I20190517-20:42:11.749(2) Meteor ➔ method subscriptions/get -> userId: guHukrC4WR8wSTqPL , arguments:  [] 
I20190517-20:42:11.764(2) Meteor ➔ publish stream-apps -> userId: guHukrC4WR8wSTqPL , arguments:  [ 'app/added', { useCollection: false, args: [] } ] 
I20190517-20:42:11.767(2) API ➔ debug GET: /api/apps/languages 
I20190517-20:42:11.772(2) API ➔ debug Success { statusCode: 200, body: { apps: [], success: true } } 
I20190517-20:42:11.775(2) Meteor ➔ publish stream-apps -> userId: guHukrC4WR8wSTqPL , arguments:  [ 'app/removed', { useCollection: false, args: [] } ] 
I20190517-20:42:11.785(2) Meteor ➔ publish stream-apps -> userId: guHukrC4WR8wSTqPL , arguments:  [ 'app/updated', { useCollection: false, args: [] } ] 
I20190517-20:42:11.788(2) Meteor ➔ publish stream-apps -> userId: guHukrC4WR8wSTqPL , arguments:  [ 'app/statusUpdate', { useCollection: false, args: [] } ] 
I20190517-20:42:11.791(2) Meteor ➔ publish stream-apps -> userId: guHukrC4WR8wSTqPL , arguments:  [ 'app/settingUpdated', { useCollection: false, args: [] } ] 
I20190517-20:42:11.794(2) Meteor ➔ publish stream-apps -> userId: guHukrC4WR8wSTqPL , arguments:  [ 'command/added', { useCollection: false, args: [] } ] 
I20190517-20:42:11.796(2) Meteor ➔ publish stream-apps -> userId: guHukrC4WR8wSTqPL , arguments:  [ 'command/disabled', { useCollection: false, args: [] } ] 
I20190517-20:42:11.799(2) Meteor ➔ publish stream-apps -> userId: guHukrC4WR8wSTqPL , arguments:  [ 'command/updated', { useCollection: false, args: [] } ] 
I20190517-20:42:11.802(2) Meteor ➔ publish stream-apps -> userId: guHukrC4WR8wSTqPL , arguments:  [ 'command/removed', { useCollection: false, args: [] } ] 
I20190517-20:42:11.816(2) Meteor ➔ method userSetUtcOffset -> userId: guHukrC4WR8wSTqPL , arguments:  [ 1 ] 
I20190517-20:42:13.060(2) Meteor ➔ method UserPresence:online -> userId: guHukrC4WR8wSTqPL , arguments:  [ null ]
```

Registering an account with the anonymous feature disabled.
```
I20190517-20:43:10.499(2) Meteor ➔ method registerUser -> userId: null , arguments:  [ { name: 'testyyy',     email: 'testyyy@testytt.com',     pass: 'testyy',     'confirm-pass': 'testyy' } ] 
I20190517-20:43:10.856(2) ====== BEGIN MAIL #8 ====== 
Mail Jargon Deleted
I20190517-20:43:10.896(2) ====== END MAIL #8 ====== 
I20190517-20:43:11.087(2) Meteor ➔ publish null -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.185(2) Meteor ➔ publish stream-importers -> userId: q57q6jezi4oPvt6da , arguments:  [ 'progress', { useCollection: false, args: [] } ] 
I20190517-20:43:11.191(2) API ➔ debug GET: /api/v1/commands.list 
I20190517-20:43:11.201(2) Meteor ➔ method autoTranslate.getSupportedLanguages -> userId: q57q6jezi4oPvt6da , arguments:  [ 'en' ] 
I20190517-20:43:11.223(2) API ➔ debug Success { statusCode: 200,   body:     { commands:        [ [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object],         [Object] ],      offset: 0,      count: 23,      total: 23,      success: true } } 
I20190517-20:43:11.245(2) Meteor ➔ method listCustomSounds -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.260(2) Meteor ➔ method listEmojiCustom -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.265(2) Meteor ➔ method apps/is-enabled -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.279(2) Meteor ➔ publish webdavAccounts -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.290(2) Meteor ➔ publish userData -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.306(2) Meteor ➔ publish activeUsers -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.310(2) Meteor ➔ method getUserRoles -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.331(2) Meteor ➔ publish userData -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.336(2) Meteor ➔ publish activeUsers -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.344(2) Meteor ➔ method rooms/get -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.364(2) Meteor ➔ method subscriptions/get -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:11.368(2) API ➔ debug GET: /api/apps/languages 
I20190517-20:43:11.370(2) API ➔ debug Success { statusCode: 200, body: { apps: [], success: true } } 
I20190517-20:43:11.395(2) Meteor ➔ publish stream-apps -> userId: q57q6jezi4oPvt6da , arguments:  [ 'app/added', { useCollection: false, args: [] } ] 
I20190517-20:43:11.398(2) Meteor ➔ publish stream-apps -> userId: q57q6jezi4oPvt6da , arguments:  [ 'app/removed', { useCollection: false, args: [] } ] 
I20190517-20:43:11.400(2) Meteor ➔ publish stream-apps -> userId: q57q6jezi4oPvt6da , arguments:  [ 'app/updated', { useCollection: false, args: [] } ] 
I20190517-20:43:11.402(2) Meteor ➔ publish stream-apps -> userId: q57q6jezi4oPvt6da , arguments:  [ 'app/statusUpdate', { useCollection: false, args: [] } ] 
I20190517-20:43:11.405(2) Meteor ➔ publish stream-apps -> userId: q57q6jezi4oPvt6da , arguments:  [ 'app/settingUpdated', { useCollection: false, args: [] } ] 
I20190517-20:43:11.407(2) Meteor ➔ publish stream-apps -> userId: q57q6jezi4oPvt6da , arguments:  [ 'command/added', { useCollection: false, args: [] } ] 
I20190517-20:43:11.410(2) Meteor ➔ publish stream-apps -> userId: q57q6jezi4oPvt6da , arguments:  [ 'command/disabled', { useCollection: false, args: [] } ] 
I20190517-20:43:11.412(2) Meteor ➔ publish stream-apps -> userId: q57q6jezi4oPvt6da , arguments:  [ 'command/updated', { useCollection: false, args: [] } ] 
I20190517-20:43:11.415(2) Meteor ➔ publish stream-apps -> userId: q57q6jezi4oPvt6da , arguments:  [ 'command/removed', { useCollection: false, args: [] } ] 
I20190517-20:43:11.423(2) Meteor ➔ method userSetUtcOffset -> userId: q57q6jezi4oPvt6da , arguments:  [ 1 ] 
I20190517-20:43:11.451(2) Meteor ➔ method getUsernameSuggestion -> userId: q57q6jezi4oPvt6da , arguments:  [] 
I20190517-20:43:12.695(2) Meteor ➔ method UserPresence:online -> userId: q57q6jezi4oPvt6da , arguments:  [ null ] 
I20190517-20:43:16.969(2) Meteor ➔ method setUsername -> userId: q57q6jezi4oPvt6da , arguments:  [ 'testyyy' ] 
I20190517-20:43:16.997(2) ====== BEGIN MAIL #9 ====== 
Mail Jargon Deleted 
I20190517-20:43:17.020(2) ====== END MAIL #9 ====== 
I20190517-20:43:17.082(2) Meteor ➔ method joinDefaultChannels -> userId: q57q6jezi4oPvt6da , arguments:  [ undefined ] 
I20190517-20:43:17.129(2) Integrations ➔ Outgoing WebHook.debug Execute Trigger: userCreated 
I20190517-20:43:17.131(2) Integrations ➔ Outgoing WebHook.debug Got the event arguments for the event: userCreated { event: 'userCreated',   user:     { _id: 'q57q6jezi4oPvt6da',      createdAt: 2019-05-17T18:43:10.614Z,      services: { password: [Object], email: [Object], resume: [Object] },      emails: [ [Object] ],      type: 'user',      status: 'online',      active: true,      _updatedAt: 2019-05-17T18:43:17.020Z,      roles: [ 'user' ],      name: 'testyyy',      lastLogin: 2019-05-17T18:43:11.056Z,      statusConnection: 'online',      utcOffset: 1,      username: 'testyyy' } } 
I20190517-20:43:17.133(2) Integrations ➔ Outgoing WebHook.debug Starting search for triggers for the room: __any 
I20190517-20:43:17.135(2) Integrations ➔ Outgoing WebHook.debug Found 0 to iterate over and see if the match the event. 
```

**Comment by Cymity** [comment]
Still the same issue with v1.1.0

This time you get stuck on an infinite loading screen when you try to enter a channel after enabling anonymous read/write.

**Comment by sampaiodiego** [comment]
are you guys able to test https://github.com/RocketChat/Rocket.Chat/pull/14717 ?
I can provide a Docker image if that makes it easy for testing. thx

**Comment by Cymity** [comment]
@sampaiodiego 
Can confirm that anonymous users are able to read channels now without any problems.
Just the registering of new accounts (registered or anonymous) doesn't work.

**Comment by MarcosSpessatto** [comment]
@Cymity can you please try again? I just updated the PR with a new commit. Thanks.

**Comment by Cymity** [comment]
@MarcosSpessatto 
Just tested it and everything seems to be working fine, no issues whatsoever. The dialog box to enter in a username pops up with the anonymous read/write feature is turned on and anonymous users are able to chat.

**Comment by MartinSchoeler** [comment]
Should be fixed by #14717 if you experience this again, please open an issue with updated details. Thanks!
